test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommandTests/RestoreCommandTests.cs

```diff
@@ -2031,14 +2045,21 @@ public async Task RestoreCommand_CentralVersion_ErrorWhenFloatingCentralVersions
 
                 var result = await restoreCommand.ExecuteAsync();
 
-                // Assert
-                Assert.False(result.Success);
-                Assert.Equal(1, logger.ErrorMessages.Count);
-                logger.ErrorMessages.TryDequeue(out var errorMessage);
-                Assert.True(errorMessage.Contains("Centrally defined floating package versions are not allowed."));
-                var messagesForNU1011 = result.LockFile.LogMessages.Where(m => m.Code == NuGetLogCode.NU1011);
-                Assert.Equal(1, messagesForNU1011.Count());
-
+                if (enabled)
+                {
+                    // Assert
```

Comment:
nitpick: IMO the assert comment can move to before the `if (enabled)` line, and the second assert comment can be removed. I think the point of the Arrange, Act, Assert pattern is to break tests into these three broad sections, rather than to be very precise about exactly when the first Assert statement is executing.